src/main/java/application/MenuController.java

```diff
@@ -322,69 +318,59 @@ private MenuSession handlePreviewCommand(SessionNavigationBean sessionNavigation
     /**
      * If we've encountered a QueryScreen and have a QueryDictionary, do the query
      * and update the session, screen, and notification message accordingly.
-     *
+     * <p>
      * Will do nothing if this wasn't a query screen.
      */
-    private NotificationMessageBean checkDoQuery(Screen nextScreen,
-                              MenuSession menuSession,
-                              Hashtable<String, String> queryDictionary) throws CommCareSessionException {
-        if(nextScreen instanceof FormplayerQueryScreen && queryDictionary != null){
-            log.info("Formplayer doing query with dictionary " + queryDictionary);
-            NotificationMessageBean notificationMessageBean = doQuery((FormplayerQueryScreen) nextScreen,
-                    queryDictionary);
-            menuSession.updateScreen();
-            nextScreen = menuSession.getNextScreen();
-            log.info("Next screen after query: " + nextScreen);
-            return notificationMessageBean;
-        }
-        return null;
+    private NotificationMessage doQuery(Screen nextScreen,
+                                        MenuSession menuSession,
+                                        Hashtable<String, String> queryDictionary) throws CommCareSessionException {
+        log.info("Formplayer doing query with dictionary " + queryDictionary);
+        NotificationMessage notificationMessage = doQuery((FormplayerQueryScreen) nextScreen,
+                queryDictionary);
+        menuSession.updateScreen();
+        nextScreen = menuSession.getNextScreen();
+        log.info("Next screen after query: " + nextScreen);
+        return notificationMessage;
     }
 
-    protected NotificationMessageBean doQuery(FormplayerQueryScreen nextScreen,
-                                              Hashtable<String, String> queryDictionary) {
+    protected NotificationMessage doQuery(FormplayerQueryScreen nextScreen,
```

Comment:
It's a bit confusing to have these both called `doQuery` when they don't do the same thing. Could we rename the first one to something like `doQueryAdvanceToNext`?

Same for `doSync`